feat(replay): Deprecate privacy API (#6174)
getsentry/sentry-docs: src/platforms/javascript/common/session-replay/privacy.mdx

| maskAllInputs    | boolean                  | `true`                                  | Mask values of `<input>` elements. Passes input values through `maskTextFn` before sending to server.                                                                                               |
| maskTextFn       | (text: string) => string | `(text) => '*'.repeat(text.length)`     | Function to customize how text content is masked before sending to server. By default, masks text with `*`.                                                                                         |
| unblock          | string[]                 | `.sentry-unblock, [data-sentry-unblock]`| Don't redact any elements that match the DOM selectors. Used to unblock specific media elements that are blocked with `blockAllMedia`.                                                     |
| unmask           | string[]                 | `.sentry-unmask, [data-sentry-unmask]`  | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked with `maskAllText`.                                                                |

### Deprecated Options

> Note that the privacy API prior to version 7.35.0 has been deprecated and replaced with the options above. Please see the [Replay migration guide](https://github.com/getsentry/sentry-javascript/blob/master/packages/replay/MIGRATION.md) for further information.

Author: ```suggestion
Note that the privacy API prior to version 7.35.0 has been deprecated and replaced with the options above. Please see the [Replay migration guide](https://github.com/getsentry/sentry-javascript/blob/master/packages/replay/MIGRATION.md#upgrading-replay-from-7340-to-7350---6645) for further information.
```

## Masking

> Masking replaces the text content with something else. The default masking behavior is to replace each character with a `*`. Elements with class name `sentry-mask` or the attribute `data-sentry-mask` will be blocked. In this example the relevant HTML code is: `<table class="sentry-mask">...</table>`:

Reviewer: ```suggestion
Masking replaces the text content with something else. The default masking behavior is to replace each character with a `*`. Elements with class name `sentry-mask` or the attribute `data-sentry-mask` will be blocked. In this example, the relevant HTML code is: `<table class="sentry-mask">...</table>`:
```

> Masking replaces the text content with something else. The default masking behavior is to replace each character with a `*`. Elements with class name `sentry-mask` or the attribute `data-sentry-mask` will be blocked. In this example, the relevant HTML code is: `<table class="sentry-mask">...</table>`:

Reviewer 2: You are writing `blocked` here, but I think you mean `masked`?

![Masking example](./session-replay-masking.png)

## Blocking

> Blocking replaces the element with a placeholder that has the same dimensions. The recording will show an empty space where the content was. Elements with class name `sentry-block` or the attribute `data-sentry-block` will be blocked. In this example the relevant HTML code is: `<table data-sentry-block>...</table>`:

Reviewer: ```suggestion
Blocking replaces the element with a placeholder that has the same dimensions. The recording will show an empty space in place of the content. Elements with class name `sentry-block` or the attribute `data-sentry-block` will be blocked. In this example, the relevant HTML code is: `<table data-sentry-block>...</table>`:
```

![Blocking example](./session-replay-blocking.png)

## Ignoring

> Ignoring only applies to form inputs. Events will be ignored on the input element so that the replay doesn't show what occurs inside of the input. Elements with class name `sentry-ignore` or the attribute `data-sentry-ignore` will have their events be ignored. In the below example, notice how the results in the table below the input changes, but no text is visible in the input:

Reviewer: ```suggestion
Ignoring only applies to form inputs. Events will be ignored on the input element so that the replay doesn't show what occurs inside of the input. Any event on an input element with class name `sentry-ignore` or the attribute `data-sentry-ignore` will be ignored. Notice how the results in the table below show input changes, but no visible text:
```

Let me know if my changes work or if they change the meaning of what we're trying to say too much.

| maskAllInputs    | boolean                  | `true`                                  | Mask values of `<input>` elements. Passes input values through `maskTextFn` before sending to server.                                                                                               |
| maskTextFn       | (text: string) => string | `(text) => '*'.repeat(text.length)`     | Function to customize how text content is masked before sending to server. By default, masks text with `*`.                                                                                         |
| unblock          | string[]                 | `.sentry-unblock, [data-sentry-unblock]`| Don't redact any elements that match the DOM selectors. Used to unblock specific media elements that are blocked with `blockAllMedia`.                                                     |
| unmask           | string[]                 | `.sentry-unmask, [data-sentry-unmask]`  | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked with `maskAllText`.                                                                |

> ### Deprecated Options

Reviewer 2: Just thinking - do we still want to show the deprecated options in the docs? What about putting them in the migration docs and linking there (or similar)?

> | unblock          | string[]                 | `.sentry-unblock, [data-sentry-unblock]`| Do not redact any elements that match the DOM selectors. Used to unblock specific media elements that are blocked due to using `blockAllMedia`.                                                     |
> | unmask           | string[]                 | `.sentry-unmask, [data-sentry-unmask]`  | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked due to using `maskAllText`.                                                                |

Reviewer: ```suggestion
| unblock          | string[]                 | `.sentry-unblock, [data-sentry-unblock]`| Don't redact any elements that match the DOM selectors. Used to unblock specific media elements that are blocked with `blockAllMedia`.                                                     |
| unmask           | string[]                 | `.sentry-unmask, [data-sentry-unmask]`  | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked with `maskAllText`.                                                                |
```

> ### Deprecated options

Reviewer: ```suggestion
### Deprecated Options
```

> In order to streamline our privacy options, the following have been deprecated in favor of the respective options above.
> 

Reviewer: ```suggestion
In order to streamline our privacy options, the following have been deprecated and replaced with the options above.

```

![Masking example](./session-replay-masking.png)

## Blocking

Blocking replaces the element with a placeholder that has the same dimensions. The recording will show an empty space where the content was. Elements with class name `sentry-block` or the attribute `data-sentry-block` will be blocked. In this example the relevant HTML code is: `<table data-sentry-block>...</table>`:
![Blocking example](./session-replay-blocking.png)

## Ignoring

Ignoring only applies to form inputs. Events will be ignored on the input element so that the replay doesn't show what occurs inside of the input. Elements with class name `sentry-ignore` or the attribute `data-sentry-ignore` will have their events be ignored. In the below example, notice how the results in the table below the input changes, but no text is visible in the input:

![Ignoring Example](./session-replay-ignore-input.gif)

## Privacy Configuration

The following options can be configured as options to the integration, in `new Replay({})`:

> The following options can be configured as options to the integration, in `new Replay({})`:

Reviewer 3: dupe sentence

| unblock          | string[]                 | `.sentry-unblock, [data-sentry-unblock]`| Do not redact any elements that match the DOM selectors. Used to unblock specific media elements that are blocked due to using `blockAllMedia`.                                                     |
| unmask           | string[]                 | `.sentry-unmask, [data-sentry-unmask]`  | Unmask all elements that match the given DOM selectors. Used to unmask specific elements that are masked due to using `maskAllText`.                                                                |

### Deprecated options
In order to streamline our privacy options, the following have been deprecated in favor of the respective options above.

| deprecated key   | replaced by | description                                                                                                                                      |
| ---------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |

> | maskInputOptions | mask        | Use CSS selectors in `mask` in order to mask all inputs of a certain type. For example, `input[type="address"]`                                  |

Author: Ah yeah you're right, it was unintentional on my part to remove it

> | maskInputOptions | mask        | Use CSS selectors in `mask` in order to mask all inputs of a certain type. For example, `input[type="address"]`                                  |

Reviewer 4: Leaving aside the breaking change, I think it's good that we don't support this option anymore fwiw. Just wanna make sure the docs are aligned (and we might wanna add a word about it in our migration doc)
